reflect.go

```diff
@@ -1018,3 +1014,34 @@ func writeReflectJSON(rv reflect.Value, rt reflect.Type, opts Options, w io.Writ
 	}
 
 }
+
+func isEmpty(rt reflect.Type, rv reflect.Value, opts Options) bool {
```

Comment:
ok. It's slower because writeReflectJSON might decide its empty sooner than the direct "reflect on every field" approach I'm taking?